### PR TITLE
Update Pipelines with Templates

### PR DESCRIPTION
**Pull request by daalcant**

### CODE REVIEW

File: Pipelines/templates/dotnet-publish-nix-job.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      archiveType: 'zip'
+      archiveFile: 'Archives/${{ parameters.projectName }}_Nix.zip'
+      replaceExistingArchive: true
+  - task: PublishBuildArtifacts@1
+    displayName: Pipeline Publish Archive
+    inputs:
+      PathtoPublish: 'Archives'
+      ArtifactName: '${{ parameters.artifactName }}'
+      publishLocation: 'Container'
+
```
Comment (gfs) [inline Pipelines/templates/dotnet-publish-nix-job.yml:93]: Whitespace

File: Pipelines/templates/dotnet-publish-nix-job.yml [hunk 2]
```diff
@@ -0,0 +1,96 @@
+parameters:
+# Job Name
+- name: jobName
+  type: string
+  default: 'publish_nix'
```
Comment (gfs) [inline Pipelines/templates/dotnet-publish-nix-job.yml:5]: Either this should be named approrpiately (nix-mac-job) or it should only do linux with a separate job that does mac

File: Pipelines/templates/dotnet-publish-win-netcore-job.yml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  - task: PublishBuildArtifacts@1
+    displayName: Pipeline Publish Archive
+    inputs:
+      PathtoPublish: 'Archives'
+      ArtifactName: '${{ parameters.artifactName }}'
+      publishLocation: 'Container'
+
+
+
+
```
Comment (gfs) [inline Pipelines/templates/dotnet-publish-win-netcore-job.yml:107]: Whitespace

File: Pipelines/templates/dotnet-test-job.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  displayName: Dotnet Test
+  pool:
+    vmImage: 'windows-latest'
+  steps:
+  - task: UseDotNet@2
+    displayName: Install Dotnet SDK
+    inputs:
+      packageType: 'sdk'
+      version: ${{ parameters.dotnetVersion }}
+  - task: NuGetToolInstaller@1
```
Comment (gfs) [inline Pipelines/templates/dotnet-test-job.yml:30]: why install nuget if we're using dotnet restore?

File: Pipelines/templates/dotnet-publish-win-netcore-job.yml [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+      version: ${{ parameters.dotnetVersion }}
+  - task: NuGetToolInstaller@1
+    displayName: Install Nuget Tool
+    inputs:
+      versionSpec: ${{ parameters.nugetVersion }}
+  - bash: 'echo "##vso[task.setvariable variable=ReleaseVersion;]$RELEASE"'
+    env:
+      RELEASE: ${{ parameters.releaseVersion }}
+    displayName: Set Release Version
+  - ${{ parameters.preBuild }}
```
Comment (gfs) [inline Pipelines/templates/dotnet-publish-win-netcore-job.yml:58]: The prebuild runs here to set the version but it looks like you set the version already on line 54.

It looks like we don't have to do any manual version trickery for the nuget ones.  I'm interested in why each works each way.

File: Pipelines/templates/dotnet-publish-nix-job.yml [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+      version: ${{ parameters.dotnetVersion }}
+  - task: NuGetToolInstaller@1
+    displayName: Install Nuget Tool
+    inputs:
+      versionSpec: ${{ parameters.nugetVersion }}        
+  - bash: 'echo "##vso[task.setvariable variable=ReleaseVersion;]$RELEASE"'
+    env:
+      RELEASE: ${{ parameters.releaseVersion }}
+    displayName: Set Release Version
+  - ${{ parameters.preBuild }}
```
Comment (gfs) [inline Pipelines/templates/dotnet-publish-nix-job.yml:58]: you set the version before you run the prebuild task that sets the version here.  Am I missing something?

**Comment by gfs** [comment]
Looks super promising!

File: Pipelines/templates/dotnet-publish-nix-job.yml (hunk 1 above)
Comment (daalcant) [reply Pipelines/templates/dotnet-publish-nix-job.yml]: Removed

File: Pipelines/templates/dotnet-publish-nix-job.yml (hunk 2 above)
Comment (daalcant) [reply Pipelines/templates/dotnet-publish-nix-job.yml:5]: Using nix as a loose term here for Unix/Linux-style systems. We could create a separate job for Mac, but this is a bit faster. Easy enough to split in the future if we see a need for it.

Comment (daalcant) [reply Pipelines/templates/dotnet-publish-nix-job.yml:5]: Not opposed to renaming the default jobName though.

File: Pipelines/templates/dotnet-publish-win-netcore-job.yml (hunk 3 above)
Comment (daalcant) [reply Pipelines/templates/dotnet-publish-win-netcore-job.yml]: Removed

File: Pipelines/templates/dotnet-test-job.yml (hunk 4 above)
Comment (daalcant) [reply Pipelines/templates/dotnet-test-job.yml:30]: Pretty sure 'dotnet restore' still leans on NuGet. This is for some past bugs we encountered where random packages weren't restoring correctly, and updating the NuGet tool before running 'dotnet restore' resolved the problem.

File: Pipelines/templates/dotnet-publish-win-netcore-job.yml (hunk 5 above)
Comment (daalcant) [reply Pipelines/templates/dotnet-publish-win-netcore-job.yml:58]: Doing this to provide multiple ways of setting the release version. Not immediately useful, but may need it in the future.

File: Pipelines/templates/dotnet-publish-nix-job.yml (hunk 2 above)
Comment (gfs) [reply Pipelines/templates/dotnet-publish-nix-job.yml:5]: I think just rename the default job. The windows one is called "windows_netcore" or whatever

File: Pipelines/templates/dotnet-publish-nix-job.yml (hunk 6 above)
Comment (daalcant) [reply Pipelines/templates/dotnet-publish-nix-job.yml:58]: Doing this to provide multiple ways of setting the release version. Not immediately useful, but may need it in the future.

File: Pipelines/templates/dotnet-publish-win-netcore-job.yml (hunk 5 above)
Comment (gfs) [reply Pipelines/templates/dotnet-publish-win-netcore-job.yml:58]: I think we only need one for now... I would choose the one that is actually being used and integrate that into the template.

File: Pipelines/templates/dotnet-publish-nix-job.yml (hunk 2 above)
Comment (daalcant) [reply Pipelines/templates/dotnet-publish-nix-job.yml]: Did this, as well as changed the file name for consistency.

File: Pipelines/templates/dotnet-publish-win-netcore-job.yml (hunk 5 above)
Comment (daalcant) [reply Pipelines/templates/dotnet-publish-win-netcore-job.yml:58]: I'll remove the release version being set within the template then.

Comment (daalcant) [reply Pipelines/templates/dotnet-publish-win-netcore-job.yml:54]: Updated

**Pull request merged**